auth/src/main/java/org/aerogear/auth/impl/UserPrincipalImpl.java

```diff
@@ -129,7 +130,8 @@ UserPrincipalImpl build() {
      */
     @Override
     public boolean hasRole(final IRole role) {
-        return roles.containsKey(role.getRoleName());
+        // TODO: to be implemented
+        throw new IllegalStateException("Not implemented");
```

Comment:
The containsKey can be used here

Reply:
I am working on that ticket, I pushed it to the PR on your fork @rachael-oregan , just testing it now.